addons/ui/org.openhab.ui.cometvisu/src/main/java/org/openhab/ui/cometvisu/backend/ChartResource.java

```diff
@@ -0,0 +1,333 @@
+			Date timeEnd, long resolution) {
+		Map<Long, ArrayList<String>> data = new HashMap<Long, ArrayList<String>>();
+		try {
+			List<String> itemNames = new ArrayList<String>();
+
+			if (item instanceof GroupItem) {
+				GroupItem groupItem = (GroupItem) item;
+				for (Item member : groupItem.getMembers()) {
+					itemNames.add(member.getName());
+				}
+			} else {
+				itemNames.add(item.getName());
+			}
+			for (String itemName : itemNames) {
+				addRrdData(data, itemName, consilidationFunction, timeBegin,
+						timeEnd, resolution);
+			}
+
+		} catch (FileNotFoundException e) {
+			// rrd file does not exist, fallback to generic persistance service
+			logger.debug("no rrd file found '{}'", (RRD_FOLDER + File.separator
+					+ item.getName() + ".rrd"));
+			return getPersistenceSeries(persistenceService, item, timeBegin,
+					timeEnd, resolution);
+		} catch (Exception e) {
+			logger.error(e.getLocalizedMessage()
+					+ ": fallback to generic persistance service");
+			return getPersistenceSeries(persistenceService, item, timeBegin,
+					timeEnd, resolution);
+		}
+		return convertToRrd(data);
+	}
+
+	private ArrayList<Object> convertToRrd(Map<Long, ArrayList<String>> data) {
+		ArrayList<Object> rrd = new ArrayList<Object>();
+		for (Long time : data.keySet()) {
+			Object[] entry = new Object[2];
+			entry[0] = time;
+			entry[1] = data.get(time);
+			rrd.add(entry);
+		}
+		return rrd;
+	}
+
+	private Map<Long, ArrayList<String>> addRrdData(
+			Map<Long, ArrayList<String>> data, String itemName,
+			ConsolFun consilidationFunction, Date timeBegin, Date timeEnd,
+			long resolution) throws IOException {
+		RrdDb rrdDb = new RrdDb(RRD_FOLDER + File.separator + itemName + ".rrd");
+		FetchRequest fetchRequest = rrdDb.createFetchRequest(
+				consilidationFunction, Util.getTimestamp(timeBegin),
+				Util.getTimestamp(timeEnd), resolution);
+		FetchData fetchData = fetchRequest.fetchData();
+//		 logger.info(fetchData.toString());
+		long[] timestamps = fetchData.getTimestamps();
+		double[][] values = fetchData.getValues();
+
+		logger.debug("RRD fetch returned '{}' rows and '{}' columns",
+				fetchData.getRowCount(), fetchData.getColumnCount());
+
+		for (int row = 0; row < fetchData.getRowCount(); row++) {
+			// change to microseconds
+			long time = timestamps[row] * 1000;
+
+			if (!data.containsKey(time)) {
+				data.put(time, new ArrayList<String>());
+			}
+			ArrayList<String> vals = data.get(time);
+			int indexOffset = vals.size();
+			for (int dsIndex = 0; dsIndex < fetchData.getColumnCount(); dsIndex++) {
+				vals.add(dsIndex + indexOffset,
+						formatDouble(values[dsIndex][row], "null", true));
+			}
+		}
+		rrdDb.close();
+
+		return data;
+	}
+
+	static String formatDouble(double x, String nanString,
+			boolean forceExponents) {
+		if (Double.isNaN(x)) {
+			return nanString;
+		}
+		if (forceExponents) {
+			return df.format(x);
+		}
+		return "" + x;
+	}
+	
+	static private String getUserDataFolder() {
```

Comment:
remove this method and use org.eclipse.smarthome.config.core.ConfigConstants.getUserDataFolder() instead.